Enable windows build in the EMBEDDED_BINS_BUILDMODE=none mode

**Screenshots & Logs**
If applicable, add screenshots to help explain your problem.
Also add any output from kubectl if applicable:
```
➜  k0s git:(main) ✗ EMBEDDED_BINS_BUILDMODE=none make k0s.exe
rm -f bindata && touch bindata
printf "%s\n\n%s\n%s\n" \
                "package assets" \
                "var BinData = map[string]struct{ offset, size int64 }{}" \
                "var BinDataSize int64 = 0" \
                > pkg/assets/zz_generated_offsets.go
/Library/Developer/CommandLineTools/usr/bin/make -C embedded-bins buildmode=none TARGET_OS=windows
make[1]: *** No rule to make target `staging/windows/bin/kubelet.exe', needed by `all'.  Stop.
make: *** [.bins.windows.stamp] Error 2
```

**Additional context**
To allow enduser precompiled binaries usage we have EMBEDDED_BINS_BUILDMODE=none settings which disables staging of any binaries with k0s apart from the code of the k0s itself.

This feature is not implemented in the build rules for the `k0s.exe` target.

--- a/.github/workflows/go.yml
+++ b/.github/workflows/go.yml
@@ -58,7 +58,7 @@ jobs:
         id: generated-bindata
         with:
           path: |
-            .bins.stamp
+            .bins.linux.stamp
             embedded-bins/staging/linux/bin/
             bindata_linux
             pkg/assets/zz_generated_offsets_linux.go

--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 
-GO_SRCS := $(shell find . -type f -name '*.go')
+GO_SRCS := $(shell find . -type f -name '*.go' -a ! -name 'zz_generated*')
 
 # EMBEDDED_BINS_BUILDMODE can be either:
 #   docker	builds the binaries in docker
@@ -30,17 +30,16 @@ endif
 .PHONY: all
 all: k0s k0s.exe
 
+zz_os = $(patsubst pkg/assets/zz_generated_offsets_%.go,%,$@)
 ifeq ($(EMBEDDED_BINS_BUILDMODE),none)
 pkg/assets/zz_generated_offsets_linux.go pkg/assets/zz_generated_offsets_windows.go:
-	rm -f bindata && touch bindata
+	rm -f bindata_$(zz_os) && touch bindata_$(zz_os)
 	printf "%s\n\n%s\n%s\n" \
 		"package assets" \
 		"var BinData = map[string]struct{ offset, size int64 }{}" \
 		"var BinDataSize int64 = 0" \
 		> $@
 else
-zz_os = $(patsubst pkg/assets/zz_generated_offsets_%.go,%,$@)
-
 pkg/assets/zz_generated_offsets_linux.go: .bins.linux.stamp
 pkg/assets/zz_generated_offsets_windows.go: .bins.windows.stamp
 pkg/assets/zz_generated_offsets_linux.go pkg/assets/zz_generated_offsets_windows.go: gen_bindata.go
@@ -50,10 +49,10 @@ pkg/assets/zz_generated_offsets_linux.go pkg/assets/zz_generated_offsets_windows
 endif
 
 k0s: TARGET_OS = linux
-k0s: pkg/assets/zz_generated_offsets_linux.go .bins.linux.stamp
+k0s: pkg/assets/zz_generated_offsets_linux.go
 
 k0s.exe: TARGET_OS = windows
-k0s.exe: pkg/assets/zz_generated_offsets_windows.go .bins.windows.stamp
+k0s.exe: pkg/assets/zz_generated_offsets_windows.go
 
 k0s.exe k0s: $(GO_SRCS)
 	CGO_ENABLED=0 GOOS=$(TARGET_OS) GOARCH=$(GOARCH) go build -ldflags='-w -s -X github.com/k0sproject/k0s/pkg/build.Version=$(VERSION) -X "github.com/k0sproject/k0s/pkg/build.EulaNotice=$(EULA_NOTICE)" -X github.com/k0sproject/k0s/pkg/telemetry.segmentToken=$(SEGMENT_TOKEN)' \

